### ISSUE TITLE
azure-arm: Add `public ip sku` parameter

### ISSUE THREAD

**Issue by CorrenSoft** [hashicorp/packer-plugin-azure#292]
Please search the existing issues for relevant feature requests, and use the
reaction feature
(https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)
to add upvotes to pre-existing requests.

#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

Under certain configurations, the packer creates a Public IP to be used by the virtual machine but is always a basic SKU, and does not exist a parameter to define a different SKU.
The request is to have that possibility.

#### Use Case(s)

As part of the process, some tools are installed and some information is retrieved from an IP-restricted site. To be able to configure an appropriate allow-list is necessary to control the outbound IP, and for that purpose, I have a NAT Gateway associated with the subnet used by Packer.
The whole process fails because Azure does allow the creation of any other public IP with the Basic Sku in the same subnet as NAT Gateway.

#### Potential configuration

HCL
```
source "azure-arm" "example" {
  ...
  public_ip_sku = "Standard"
  ...
}
```

**Comment by plessisa** [comment]
This will soon become more critical since Microsoft is deprecating the Public IP Address Basic SKU in 2025.
https://azure.microsoft.com/en-us/updates/upgrade-to-standard-sku-public-ip-addresses-in-azure-by-30-september-2025-basic-sku-will-be-retired/

**Comment by JenGoldstrich** [comment]
Hey friends, I've merged a PR #416 to add this requested functionality, it also includes a warning to let users know to try out standard SKU Ip addresses so that users try this before the deprecation, in a future release we will shift to using Standard as the default after we give users some time to try it out and catch any issues with this functionality long before the March 2025 deadline (in March you can't make new IPs anymore on the old SKU so this would break for the ARM builder)
